### PR TITLE
Wait for form elements

### PR DESCRIPTION
**Pull request by mfalesni**
YOu can either specify True to use the element and wait for it or you can specify a custom locator to be waited for.

{{pytest: cfme/tests/cloud_infra_common/test_events.py -v -k vm_create --use-provider complete }}

### CODE REVIEW

**Comment by psav** [comment]
Shuldn't we just put wait_loc on everything? ie...inside the fill have it do a locate then an is_displayed?

**Comment by dajoRH** [comment]
Lint report for commit c0607245cd1b8ddd8305143e0baec004baf61797:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav not all "object" that we put in the forms can locate() :sob:

**Comment by dajoRH** [comment]
Lint report for commit b84f780d927a3cc002794e0b92d80efb840c9d18:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav updated.

**Comment by dajoRH** [comment]
Lint report for commit 8295c7cdbbc19f535ee5fa97ea7a2aca0f5d7b8d:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a398ab770fa8f20275c50931acae4915e04120fc:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/web_ui/__init__.py [hunk 1]
```diff
@@ -1485,6 +1485,10 @@ def _fill_form_list(form, values, action=None, action_always=False):
     for field, value in values:
         if value is not None and form.field_valid(field):
             loc = form.locators[field]
+            try:
+                sel.wait_for_element(loc)
+            except TypeError:
+                logger.debug(' It was not possible to wait for %r because it is not locatable', loc)
```
Comment (psav) [inline cfme/web_ui/__init__.py]: We were supposed to have a ```sleep 1``` in here no?

Comment (mfalesni) [inline cfme/web_ui/__init__.py]: Damn, yes, thanks.

**Comment by dajoRH** [comment]
Lint report for commit d07e0844b76224a94f2a0d11aab9fdac574273a9:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cf5427863684b03ca545dea750ee335e1f2e2cc2:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0cf930b42a96e0751271202e2bd5483fcdff530e:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit df4dee5e392d99d1637878264b15a6e0a60cbcfd:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b66cb81e18c8b031500de1a955f407127e410bf8:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**